Sources/BSON/BSON.swift

```diff
@@ -77,6 +77,42 @@ public enum BSON {
         }
     }
 
+    /// Initialize a `BSON` from ExtendedJSON
+    /// Parameters:
+    ///   - `json`: a `JSON` representing the canonical or relaxed form of ExtendedJSON for any `BSONValue`.
+    ///   - `keyPath`: an array of `Strings`s containing the enclosing JSON keys of the current json being passed in.
+    ///              This is used for error messages.
+    ///
+    /// Throws:
+    ///   - `InternalError`
```

Comment:
Generally we don't call out when `InternalError`s are thrown, since they theoretically shouldn't ever occur. We should call out that this will throw `DecodingError`s when malformatted extjson is passed in though.